charmcraft.yaml

```diff
@@ -15,39 +15,13 @@ links:
   source: https://github.com/canonical/grafana-agent-operator
   issues: https://github.com/canonical/grafana-agent-operator/issues
 
-bases:
-  - build-on:
-    - name: "ubuntu"
-      channel: "22.04"
-      architectures: ["arm64"]
-    run-on:
-    - name: "ubuntu"
-      channel: "22.04"
-      architectures: ["arm64"]
-  - build-on:
-    - name: "ubuntu"
-      channel: "20.04"
-      architectures: ["arm64"]
-    run-on:
-    - name: "ubuntu"
-      channel: "20.04"
-      architectures: ["arm64"]
-  - build-on:
-    - name: "ubuntu"
-      channel: "22.04"
-      architectures: ["amd64"]
-    run-on:
-    - name: "ubuntu"
-      channel: "22.04"
-      architectures: ["amd64"]
-  - build-on:
-    - name: "ubuntu"
-      channel: "20.04"
-      architectures: ["amd64"]
-    run-on:
-    - name: "ubuntu"
-      channel: "20.04"
-      architectures: ["amd64"]
+platforms:
+  ubuntu@20.04:amd64:
+  ubuntu@20.04:arm64:
+  ubuntu@22.04:amd64:
+  ubuntu@22.04:arm64:
+  ubuntu@24.04:amd64:
+  ubuntu@24.04:arm64:
```

Comment:
@lucabello: on amd64, a pack produces the following files:
```
grafana-agent_ubuntu@20.04-amd64.charm
grafana-agent_ubuntu@22.04-amd64.charm
grafana-agent_ubuntu@24.04-amd64.charm
```

Are we ok CI-wise with this? For arm64 too?

Reply:
Yes, we haven't had a case where these many charm files are produced, but this should work with new CI out-of-the-box :)

`amd`+`arm` I've seen running, multiple-base on the same architecture I haven't, but CI should loop over all the charms, so we should be good